### ISSUE TITLE
When testing HttpApi in local, queryStringParameters should be either undefined or non-empty

### ISSUE THREAD

**Issue by mastazi** [aws/aws-sam-cli#3081]
### Description:

I have this function:
```
fakeFunction:
    Type: AWS::Serverless::Function
    Properties:
      Handler: src/handlers/handlers.fake
      Description: A Lambda function.
      Events:
        Api:
          Type: HttpApi
          Properties:
            Path: /my-path/
            Method: get
```
When I test this in local, even if I call the function without any queryString parameters, the `event` object  still has the property `queryStringParameters`. However, after deployment, the property is not created at all.

so as an example, if my function was in Node.js, attempting this check: `if (event.queryStringParameters.hasOwnProperty('myproperty')){...}` will work in SAM local, but after deployment will result in the error “cannot read property ‘myproperty’ of undefined”. 

the only way it would work after deployment, is this:
`if (event.hasOwnProperty('queryStringParameters') && event.queryStringParameters.hasOwnProperty('myproperty'))`

I think the emulator should work in the same way as "real world" Api Gateway, so `queryStringParameters` should be either undefined or non-empty.

### Steps to reproduce:

1. create a function defined as in the `template.yml` snippet above, then include this check in the function:
```
if (event.queryStringParameters.hasOwnProperty('myproperty')) {
  console.log("checking)
}
``` 
2. run `sam local start-api` and call the function without any query string parameters. It will work.
3. deploy the function, then call it without any query string parameters. You will get 500 error.

### Observed result:

Go to CloudWatch and check the function logs after completing step 3 above.
In CloudWatch you will find an error message such as the one below:

```
{
    "errorType": "TypeError",
    "errorMessage": "Cannot read property 'hasOwnProperty' of undefined",
    "stack": [
        "TypeError: Cannot read property 'hasOwnProperty' of undefined",
        "    at Runtime.exports.fakeFunction [as handler] (/var/task/src/handlers/handlers.js:43:37)",
        "    at processTicksAndRejections (internal/process/task_queues.js:95:5)"
    ]
}
```
However there will be no error when executing the function locally, as observed at point 2 above.

### Expected result:

The expected result is that the same error "Cannot read property 'hasOwnProperty' of undefined" should also be thrown in local simulator.

### Additional environment details (Ex: Windows, Mac, Amazon Linux etc)

1. OS: Windows 10
2. `sam --version`: SAM CLI, version 1.23.0
3. AWS region: ap-southeast-2

**Comment by jfuss** [comment]
@mastazi Thanks for the report. We do try to replicate what we can from the "real" API Gateway. It looks like this might be caused by https://github.com/aws/aws-sam-cli/blob/234655e4cbc963e14a5e078b36902bd7a0498c70/samcli/local/events/api_event.py#L411. Compare this to what what we do for Rest: https://github.com/aws/aws-sam-cli/blob/234655e4cbc963e14a5e078b36902bd7a0498c70/samcli/local/events/api_event.py#L231. So we might need the same logic in the V2 version. 

Updating labels

**Comment by mastazi** [comment]
@jfuss thanks for identifying the code responsible for it, yes I agree looks like the `if ... else None` will do the trick here.

**Comment by tomatod** [comment]
This seems has been fixed in [#3497](https://github.com/aws/aws-sam-cli/pull/3497).
I did not reproduce this issue #3018 with version 1.40.0.

**Comment by super132** [comment]
Since PR #3497 fixed the issue, I'm closing it now.
